fiona/ogrext.pyx

```diff
@@ -775,7 +775,11 @@ cdef class Session:
             raise ValueError("Null layer")
 
         result = OGR_L_GetExtent(self.cogr_layer, &extent, 1)
-        return (extent.MinX, extent.MinY, extent.MaxX, extent.MaxY)
+        
+        if result == OGRERR_NONE:
+            return (extent.MinX, extent.MinY, extent.MaxX, extent.MaxY)
+        else:
+            raise DriverError("Driver was not able to calculate extent")
```

Comment:
It's only a matter of style, but my preference is to indent the errors and keep the happy path unindented (as far as possible). This also means the expected return is at the end of the function. e.g.:
```suggestion
        if result != OGRERR_NONE:
            raise DriverError("Driver was not able to calculate extent")
        return (extent.MinX, extent.MinY, extent.MaxX, extent.MaxY)
```